endpoints/openrtb2/auction.go

```diff
@@ -491,23 +491,35 @@ func fillAndValidateNativeAssets(assets []nativeRequests.Asset, impIndex int) er
 		return fmt.Errorf("request.imp[%d].native.request.assets must be an array containing at least one object", impIndex)
 	}
 
+	assetIDs := make(map[int64]bool, len(assets))
+	assignAssetIDs := (assets[0].ID == 0)
```

Comment:
The assumption here is that `0` signifies an unset ID and if the first asset has an unset ID then Prebid Server should assign ids. The problem is an ID of 0 for the 0th item makes sense if you're following the recommendation of using array index values as IDs. Instead, I propose we set `assignAssetIDs` to `true` when every asset has an ID if 0.

Reply:
good point, ill update